apps/web/app/environments/[environmentId]/settings/product/editProduct.tsx

```diff
@@ -91,3 +99,58 @@ export function EditWaitingTime({ environmentId }) {
     </form>
   );
 }
+
+export function DeleteProduct({ environmentId }) {
+  const router = useRouter();
+
+  const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false)
+
+  const { product, isLoadingProduct, isErrorProduct } = useProduct(environmentId);
+  const { environment } = useEnvironment(environmentId);
+
+  if (isLoadingProduct) {
+    return <LoadingSpinner />;
+  }
+  if (isErrorProduct) {
+    return <ErrorComponent />;
+  }
+
+  const handleDeleteProduct = async () => {
+    if (environment.availableProducts?.length <= 1) {
+      toast.error("Cannot delete product. Environment needs at least 1.");
+      setIsDeleteDialogOpen(false);
+      return;
+    }
+    const deleteResponse = await deleteProduct(environmentId);
+
+    if (deleteResponse?.message?.length > 0) {
+      toast.error(deleteResponse.message);
+      setIsDeleteDialogOpen(false);
+    }
+    if (deleteResponse?.id?.length > 0) {
+      toast.success("Product deleted successfully.");
+      router.push("/environments");
+    }
+  }
+
+  return (
+    <div>
+      <p className="text-sm text-slate-900">
```

Comment:
I think it would be great user experience if you also have a check here for the current user roles and permissions to delete a product and if the environment has more than 1 product. 
For example, you can disable and gray out the Delete button and add some text below why it's disabled.

Reply:
Yes, it will be good UX. However, we are checking these validations.

But as you mentioned I have disabled the delete button when any od these conditions are met. 
And there will be an error message below the disabled delete button informing the user about the reason.